src/Symfony/Component/Lock/Lock.php

```diff
@@ -70,6 +71,9 @@ public function acquire($blocking = false)
     {
         try {
             if ($blocking) {
+                if (!($this->store instanceof BlockingStoreInterface && $this->store->supportsWaitAndSave())) {
+                    throw new NotSupportedException();
```

Comment:
Should we add a message to the exception?

Reply:
done